README.md

```diff
@@ -28,4 +28,263 @@ One of the very common use cases of WebXR is 180 and/or 360 photos or videos, bo
+  }
+}
+```
+
+In some cases, such as a mono `XRQuadLayer` being shown on a stereo device, multiple `XRView`s may return the same `XRSubImage` values. To avoid rendering the same view multiple times in these scenarios the `primary` attribute of the `XRSubImage` will be be set to `false` for all but one of the overlapping views. (It should be noted that `XRProjectionLayer`s will never return `XRSubImage`s with `primary` set to false.)
+
+```js
+// Render Loop for a projection layer with a WebGL framebuffer source.
+const xrGfx = new XRWebGLGraphicsBinding(xrSession, gl);
+
+const quadLayer = xrGfx.requestQuadLayer({ pixelWidth: 512, pixelHeight: 512, stereo: false });
+// Position 2 meters away from the origin with a width and height of 1.5 meters
+quadLayer.referenceSpace = xrReferenceSpace;
+quadLayer.transform = new XRRigidTransform({z: -2});
+quadLayer.width = 1.5;
+quadLayer.height = 1.5;
+
+xrSession.updateRenderState({ layers: [quadLayer] });
+xrSession.requestAnimationFrame(onXRFrame);
+
+function onXRFrame(time, xrFrame) {
+  xrSession.requestAnimationFrame(onXRFrame);
+
+  gl.bindFramebuffer(gl.FRAMEBUFFER, xrGfx.getCurrentFramebuffer(quadLayer));
+  for (let view in xrViewerPose.views) {
+    let subImage = quadLayer.getViewSubImage(view);
+
+    if (subImage.primary) {
+      gl.viewport(viewport.x, viewport.y, viewport.width, viewport.height);
+      
+      // Render content for view.eye
+    }
+  }
+}
+```
+
+## Appendix A: Proposed IDL
+
+```webidl
+//
+// Layer interface
+//
+
+interface XRLayer {
+  XRSubImage? getViewSubImage(XRView view);
+
+  readonly attribute unsigned long pixelWidth;
+  readonly attribute unsigned long pixelHeight;
+  readonly attribute unsigned long arraySize;
+  readonly attribute boolean ignoreDepthValues;
+
+  boolean blendTextureSourceAlpha = false;
+  boolean chromaticAberrationCorrection = false;
+}
+
+interface XRSubImage {
+  readonly attribute XRViewport viewport;
+  readonly attribute unsigned long imageIndex;
+  readonly attribute boolean primary;
+}
+
+//
+// Layer types
+//
+
+interface XRProjectionLayer extends XRLayer {
+}
+
+interface XRQuadLayer extends XRLayer {
+  attribute XRRigidTransform transform;
+  attribute float width = 1;
+  attribute float height = 1;
+}
+
+interface XRCylinderLayer extends XRLayer {
+  XRReferenceSpace referenceSpace;
+  attribute XRRigidTransform transform;
+  attribute float radius = 1;
+  attribute float centralAngle = Math.PI * 0.5;
+  attribute float aspectRatio = 1;
+}
+
+interface XREquirectLayer extends XRLayer {
+  XRReferenceSpace referenceSpace;
+  attribute XRRigidTransform transform;
+  attribute float radius = 1;
+  attribute float scaleX = 1;
+  attribute float scaleY = 1;
+  attribute float biasX = 0;
+  attribute float biasY = 0;
+}
+
+interface XRCubeLayer extends XRLayer {
+  XRReferenceSpace referenceSpace;
+  attribute DOMPoint orientation;
+}
+
+//
+// Graphics Bindings
+//
+
+dictionary XRWebGLLayerSourceInit {
```

Comment:
Should this just be `: XRWebGLLayerInit`?

Reply:
No, because that's an existing dictionary with attribs that don't quite match what's needed here. Please note that I think this is a terrible name, by the way, and we can definitely do something cleaner, I just haven't figured out what yet.